File: Problems/No.559_Maximum_Depth_of_N_ary_Tree/maximum_depth_of_n_ary_tree.py
"""
# Definition for a Node.
class Node(object):
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""

class Solution(object):
    def maxDepth(self, root):
        """
        :type root: Node
        :rtype: int
        """

        # parameters
        TNodes = []
        MaxDepth = 0

        # start computing
        if (root == None):
            return 0
        TNodes.append(root)
        while TNodes:
            LevelSize = len(TNodes)
            for i in range(LevelSize):
                currNode = TNodes.pop(0)
                for child in currNode.children:
                    TNodes.append(child)
            MaxDepth += 1

        return MaxDepth        

File: Problems/No.559_Maximum_Depth_of_N_ary_Tree/maximum_depth_of_n_ary_tree.cpp
/*
// Definition for a Node.
class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val) {
        val = _val;
    }

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
public:
    int maxDepth(Node* root) {
        
        // parameters
        std::queue<Node *> TNodes;
        int MaxDepth = 0;

        // start computing
        if (root == nullptr) return 0;
        TNodes.push(root);
        while (!TNodes.empty()) {

            int LevelSize = TNodes.size();
            for (int i=0; i<LevelSize; i++) {

                auto currNode = TNodes.front();
                TNodes.pop();
                for (auto child: currNode->children) TNodes.push(child);

            }
            
            MaxDepth++;

        }

        return MaxDepth;

    }
};
